### PR TITLE
Bazel 8 release

### PR DESCRIPTION
**Pull request by keertk**
(not ready to be merged yet)

### CODE REVIEW

File: _posts/2024-12-09-bazel-8-release.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+- **Shell:** All sh_* rules are now part of [rules_shell](https://github.com/bazelbuild/rules_shell).
+
+To ease this transition, Bazel 8.0 introduces the `--incompatible_autoload_externally` flag which automatically loads rules from their respective repositories, thus removing the need for an imminent migration. The migration to use load statements needs to happen in 2025 before Bazel 9 is released. Load statements are supported on Bazel versions 6, 7 and 8, so the migration is largely independent of the Bazel version.
+
+### Bzlmod: the new default for dependency management
+
+Bzlmod was introduced in Bazel 5 in 2021 as an experimental feature, it became GA in Bazel 6 and turned on in Bazel 7. In Bazel 8, the new powerful external dependencies system works with the old WORKSPACE mechanism turned off by default. This shift means that Bazel no longer automatically reads the WORKSPACE and WORKSPACE.bzlmod files when building your project. These files were previously required for defining external repositories and dependencies, but Bzlmod offers a more modern and efficient approach.
```
Comment (meisterT) [inline _posts/2024-12-09-bazel-8-release.md:34]:
```suggestion
Bzlmod was introduced in Bazel 5 in 2021 as an experimental feature, it became GA in Bazel 6 and turned on in Bazel 7. In Bazel 8, the new powerful external dependencies system works with the old WORKSPACE mechanism turned off by default. This shift means that Bazel no longer automatically reads the `WORKSPACE` and `WORKSPACE.bzlmod` files when building your project. These files were previously required for defining external repositories and dependencies, but Bzlmod offers a more modern and efficient approach.
```

File: _posts/2024-12-09-bazel-8-release.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+We encourage all users to migrate to Bzlmod and take advantage of its benefits. Take a look at this [migration guide](https://bazel.build/external/migration) for step-by-step instructions and best practices for moving from the traditional WORKSPACE setup to the streamlined Bzlmod system.
+
+We are aiming to remove all WORKSPACE functionalities in Bazel 9.
+
+### Symbolic macros
+
+Symbolic macros offer a new way to write macros that are safer to use and that catch bugs sooner. For example, their arguments are typed like rule attributes, and the author decides which attributes permit select()s based on whether the macro needs to examine that value or pass it through. With Symbolic macros come several related features:
+
+- A new macro-aware visibility model, which protects a macro's internal targets from being depended on by its caller
+- Rule Finalizers, which are macros that can call native.existing_rules() with less surprising behavior
```
Comment (meisterT) [inline _posts/2024-12-09-bazel-8-release.md:51]:
```suggestion
- Rule Finalizers, which are macros that can call `native.existing_rules()` with less surprising behavior
```

**Comment by meteorcloudy** [comment]
@keertk Not sure what's going on with the CLA check, maybe you can try to squash the commits and make sure no extra git author was recorded in the commit message.

**Comment by keertk** [comment]
Yeah I'm not sure what's going on - I just accepted / committed Tobi's suggestions directly here.

**Comment by keertk** [comment]
https://github.com/bazelbuild/bazel-blog/pull/362

**Pull request closed without merging**